ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                                 (size_t)FlushInfo->nNumBytes, FlushInfo->pF))
+        return 1;
+    FlushInfo->OffsetWhereToFlush += FlushInfo->nNumBytes;
+    FlushInfo->NTimesFlushed++;
+    FlushInfo->TotalSavedBytes += FlushInfo->nNumBytes;
+    FlushInfo->nNumBytes = 0;
+
+    return 0;
+}
+
+// Reads a block of data from a buffer in memory
+int MMReadBlockFromBuffer(struct MM_FLUSH_INFO *FlushInfo)
+{
+    if (!FlushInfo->SizeOfBlockToBeSaved)
+        return 0;
+
+    if (FlushInfo->pBlockToBeSaved)
+    {
+        memcpy(FlushInfo->pBlockToBeSaved,
+               (void *)((char *)FlushInfo->pBlockWhereToSaveOrRead +
+                        FlushInfo->CurrentOffset),
+               FlushInfo->SizeOfBlockToBeSaved);
+    }
+    FlushInfo->CurrentOffset += FlushInfo->SizeOfBlockToBeSaved;
+
+    return 0;
+}
+
+// Appends a block of data to a buffer in memory, which is
+// used for later flushing to disk.
+int MMAppendBlockToBuffer(struct MM_FLUSH_INFO *FlushInfo)
+{
+    if (FlushInfo->SizeOfBlockToBeSaved)
+    {
+        // If all the bloc itselfs does not fit to the buffer,
+        // then all the block is written directly to the disk
+        if (FlushInfo->nNumBytes == 0 &&
+            FlushInfo->SizeOfBlockToBeSaved >= FlushInfo->nBlockSize)
+        {
+            if (MMFlushToDisk(FlushInfo))
+                return 1;
+            return 0;
+        }
+
+        // There is space in FlushInfo->pBlockWhereToSaveOrRead?
+        if (FlushInfo->nNumBytes + FlushInfo->SizeOfBlockToBeSaved <=
+            FlushInfo->nBlockSize)
+        {
+            if (FlushInfo->pBlockToBeSaved)
+            {
+                memcpy((void *)((char *)FlushInfo->pBlockWhereToSaveOrRead +
+                                FlushInfo->nNumBytes),
+                       FlushInfo->pBlockToBeSaved,
+                       FlushInfo->SizeOfBlockToBeSaved);
+            }
+            else  // Add zero caracters
+            {
+                char zero_caracters[8] = "\0\0\0\0\0\0\0";
+                memcpy((char *)FlushInfo->pBlockWhereToSaveOrRead +
+                           FlushInfo->nNumBytes,
+                       zero_caracters, FlushInfo->SizeOfBlockToBeSaved);
+            }
+
+            FlushInfo->nNumBytes += FlushInfo->SizeOfBlockToBeSaved;
+        }
+        else
+        {
+            // Empty the buffer
+            if (MMFlushToDisk(FlushInfo))
+                return 1;
+            // Append the pendant bytes
+            if (MMAppendBlockToBuffer(FlushInfo))
+                return 1;
+        }
+        return 0;
+    }
+    // Just flush to the disc.
+    return MMFlushToDisk(FlushInfo);
+}
+
+// Copy the contents of a temporary file to a final file.
+// Used everywhere when closing layers.
+int MMMoveFromFileToFile(FILE_TYPE *pSrcFile, FILE_TYPE *pDestFile,
+                         MM_FILE_OFFSET *nOffset)
+{
+    size_t bufferSize = 100 * 1024 * 1024;  // 100 MB buffer;
```

Comment:
1 MB should be more than enough and perform equally well:
```suggestion
    size_t bufferSize = 1024 * 1024;  // 1 MB buffer;
```

Reply:
Image a 500 MB file to be copied from one file to another using fread/fwrite.
Are you sure that it's the same to read/write 5 times a buffer of 100Mb than read/write 500 times a buffer of 1 MB?
We've always been taking care of that kind of things.
When we operate with small files, it doesn't matter but with large files I think it could be worth using 100Mb instead of 1Mb.
What do you think?

Reply:
And belive me, we manage long files

Reply:
I strongly doubt you such a large buffer to copy files. 1 MB should be more than enough. You can test if you want. I doubt you can measure a significant gain by increasing the buffer size. GDAL's CopyFile method only uses a 40 KB buffer: https://github.com/OSGeo/gdal/blob/35c434527aab4acb86af66480db00d4fec4b5400/port/cpl_vsil.cpp#L1347

Reply:
I've tested it in a not basic example and you're right. I had it misunderstood then. I've not used it a lot but I though that it was better less iterations with more a bigger buffer than more iterations with a small buffer.
Thanks.